restmanager/frontend/views.py

```diff
@@ -1,44 +1,142 @@
-from django.shortcuts import render
 import os
+import json
 import slack
+import requests
 from dotenv import load_dotenv
-from django.http import HttpResponse
-from . import strings
+from fridges.models import Fridge
+from django.shortcuts import render
 from django.views.decorators.csrf import csrf_exempt
+from django.http import HttpResponse, HttpResponseRedirect
 
-# Create your views here.
+# Gets slack token from .env file and sets it here
+load_dotenv()
+slack_client = slack.WebClient(token=os.getenv("SLACK_TOKEN"))
 
-# Defining what html file to return when this function is called.
+# IP2 is the servers IP Address set in .env found in fridge_manager/restmanager
+IP2 = os.getenv('IP2', "127.0.0.1")
+# D_PORT is set only here, determines which port the get_request is sent to (the port that the server is running on)
 
+D_PORT = os.getenv('D_PORT', "8100")
 
-def fridge2(request):
-    return render(request, 'frontend/fridge2.html')
 
+@csrf_exempt
+def create_floor_list():
+    """ Creates a list of unique floors found in fridges table in the database and returns it """
+    floor_list = []
 
-def manage(request):
-    return render(request, 'frontend/index.html')
+    # adds unique floor numbers found in the get_request()
+    for item in get_request():
+        if item['floor'] not in floor_list:
+            floor_list.append(item['floor'])
+    # sorts the floor_list from smallest to largest so the list makes sense in the view
+    floor_list.sort()
+    return floor_list
 
 
-# GET SLACK TOKEN HERE
-load_dotenv()
-client = slack.WebClient(token=os.getenv("SLACK_TOKEN"))
+@csrf_exempt
+def get_request():
+    """ Sends get request to api endpoint /api/fridges/ which returns a json object with all the fridges in
+    the database """
+    r = requests.get('HTTP://' + IP2 + ':' + D_PORT + '/api/fridges/?format=json')
+    return json.loads(r.text)
 
 
 @csrf_exempt
-def post_beer(request):
-    if request.method == "POST":
-        client.chat_postMessage(
-            channel=strings.CHANNEL_NAME_1,
-            text=strings.SLACK_MESSAGE_2
-        )
-    return HttpResponse("ok")
+def create_list(request):
+    """ Creates a list and sorts through it depending on what parameters are given in the request that it receives
+     Accepts parameters from url id, floor, state and returns a select_list named list """
+    select_list = []
+    floor = request.GET.get('floor')
+    fridge_id = request.GET.get('id')
+    state = request.GET.get('state')
+
+    # loops through data from get_request()
+    for item in get_request():
+        # if the request has a given parameter floor it is added to select_list
+        if floor is not None:
```

Comment:
Tää koko funkkari tässä on aika copy paste tän olisi voinut muuttaa esim muotoon
if floor is not None:
listAppend(floor)
elif fridge_id is not None:
listAppend(fridge_id)
elif state is not None: 
listAppend(state)
else:
  select_list.append(item)


def listAppend:
int_floor = int(id)
            if item['id'] == int_floor:
              return item

Tms. näin pseudona, anyway niin että tätä if parsintaa ei tarvii mennä näin läpi ja koittaa pohtia mitä tässä oikein koitetaan tehdä kun kaikki on saman copy/pastee lähes tulkoon.
Lisäksi pohtisin voiko tuon none checkin tehdä paremmin